src/datasets/vision/imagenet.jl

```diff
@@ -0,0 +1,155 @@
+const IMAGENET_WEBSITE = "https://image-net.org/"
+
+function __init__imagenet()
+    DEPNAME = "ImageNet"
+    return register(
+        ManualDataDep(
+            DEPNAME,
+            """The ImageNet 2012 Classification Dataset (ILSVRC 2012-2017) can be downloaded at
+            $(IMAGENET_WEBSITE) after signing up and accepting the terms of access.
+            It is therefore required that you download this dataset manually.
+
+            Please follow the instructions at
+            https://github.com/JuliaML/MLDatasets.jl/blob/master/docs/src/datasets/imagenet_installation.md.
+            """,
+        ),
+    )
+end
+
+"""
+    ImageNet(; Tx=Float32, split=:train, dir=nothing)
+    ImageNet([Tx, split])
```

Comment:
there are many keyword arguments that need to be documented

Reply:
Updated in 944bd83.